### DISCUSSION TITLE
Google Oauth2.0 setup

### DISCUSSION THREAD

**Discussion by githublj** [netbox-community/netbox#9926] [Q&A]
Hi all,
Of course did pre-requisite archive and google searches, and do feel like a complete answer is still missing.

I'm trying to setup Netbox with Google Oauth2.0 , the other most relevant and recent post ( https://github.com/netbox-community/netbox/discussions/9078 ) had a configuration example, but it was unclear if the poster was suggesting you fudge the secret/key with xxxx, or maybe it was simply implied you needed to get those values yourself.

It didn't seem overly tedious to just get a secret/key pair from the Google Cloud API, I went down that road and as soon as I had a key, my oauth started working immediately.  Whee... but there is a catch... more on that in a second.

Also, this site was interesting ( https://questionable.one/post/secure-netbox-with-oauth2/ ) , but I found it after my Google Oauth was already working.   So, I'm intrigued and confused about all the work that is shown setting up Nginx, vouch-proxy, okta, etc.  It seems far more complicated than what worked for me, right out of the figurative box?

So, back to Google and my 'catch'...  I walked through Google's documentation (and i've done it in the distant past, so it was semi-familiar): https://support.google.com/googleapi/answer/6158849?hl=en

However, Google really views everything you do as you were an original "developer" and that you are making a whole new application from scratch.  To that end, before you can use the API unrestricted you have to publish your application, and they have some requirements that seem a bit over the top, and they perform a multi-day manual verification:

![image](https://user-images.githubusercontent.com/7999228/182977166-f13d1df8-fc7f-433b-bb94-8b5f99786512.png)

And at that point, I started second guessing myself...  Did I take a wrong turn?  Shouldn't there be an 'easy way' to get a Google oauth2 secret/key pair for 'simple' authentication?  More googling, and I couldn't find anything simpler, all roads led back to the developer API.   I can and will fulfill Google's requests to get 'my app' published, but just hoping to make it easier for someone else, particularly if anyone else has setup Netbox with Google Oauth2.0 and would care to share their experience?

In summary, on the plus side, Netbox configuration while not as simple as the previous poster implied, does seem to be fairly simple and functions well.  On the negative side, Google's requirements are slightly tedious, unless there is a simpler way anyone is aware of?

And I don't think it matters much... but I installed NetBox 3.2.7 on a Ubuntu 20.04.01 AWS instance, as noted, with Nginx.  Very vanilla setup.

Thanks in advance for any responses.
